src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.tsx

```diff
@@ -47,9 +47,30 @@ export default class ThirdPartyCollectionDetailPage extends React.PureComponent<
     isAuthModalOpen: false
   }
 
+  componentDidMount() {
```

Comment:
Not related to the issue, I'm adding this here as a driveby. We should also fetch the available slots on the `componentDidMount` and not only in the `componentDidUpdate` as it's doing atm. 
The `didMount` is necessary when the user comes from the collection list, in that case, the `didUpdate` is not triggered and so the `onFetchAvailableSlots` is never called.